### PR TITLE
Fix #728 - Filter compilation output during testing

### LINKED ISSUE #728: Intellij IDEA shows errors with the sources used during testing
Reported by Slava Kornienko:

image(1)

It's confusing because it finishes successfully

### PR DESCRIPTION
**Pull request by rachelcarmena**
Closes #728 thanks to the idea by @Starmel :tada: 

It filters the compilation output during testing when avoiding lines starting with "e:"

### CODE REVIEW

**Comment by Starmel** [comment]
Modify stream on the fly without temp files:

```kotlin
// To Idea gradle task output bug work-around https://github.com/arrow-kt/arrow-meta/issues/728
class KotlincErrorSuppressProxy(stream: OutputStream) : PrintStream(stream) {

  override fun write(buf: ByteArray, off: Int, len: Int) {
    val newLine = String(buf, off, len)
      .run { if (indexOf("e:") == 0 || indexOf("w:") == 0) "kotlinc_$this" else this }
      .toByteArray()

    super.write(newLine, 0, newLine.size)
  }
}
```

```kotlin
..
messageOutputStream = KotlincErrorSuppressProxy(System.out)
..
```

**Comment by rachelcarmena** [comment]
> Modify stream on the fly without temp files:

Good idea @Starmel :raised_hands: 

Please, I changed your proposal slightly, let me know your feedback. Thanks!!

File: meta-test/src/main/kotlin/arrow/meta/plugin/testing/Compilation.kt [hunk 1]
```diff
@@ -15,6 +16,15 @@ internal fun compile(data: CompilationData): Result =
     classpaths = data.dependencies.map { classpathOf(it) }
     pluginClasspaths = data.compilerPlugins.map { classpathOf(it) }
     compilerPlugins = data.metaPlugins
+    messageOutputStream = object : PrintStream(System.out) {
+      override fun write(buf: ByteArray, off: Int, len: Int) {
+        val newLine = String(buf, off, len)
+          .run { replace(Regex("^e:"), "error found:") }
```
Comment (Starmel) [inline meta-test/src/main/kotlin/arrow/meta/plugin/testing/Compilation.kt:22]: This compiling the regular expression for every line. May reduce performance. FIY
```suggestion
    messageOutputStream = object : PrintStream(System.out) {
      
      private val kotlincErrorRegex = Regex("^e:") 
      
      override fun write(buf: ByteArray, off: Int, len: Int) {
        val newLine = String(buf, off, len)
          .run { replace(kotlincErrorRegex, "error found:") }
```

**Review by Starmel — APPROVED** [review]

File: meta-test/src/main/kotlin/arrow/meta/plugin/testing/Compilation.kt (hunk 1 above)
Comment (rachelcarmena) [reply meta-test/src/main/kotlin/arrow/meta/plugin/testing/Compilation.kt]: Good idea @Starmel :raised_hands: Thank you so much!!

**Review by danimontoya — APPROVED** [review]
Great, thank you for fixing it guys! 🙌

**Comment by rachelcarmena** [comment]
Thanks for the review @danimontoya :raised_hands: And thanks @Starmel again :tada:

**Pull request merged**